aya/src/programs/links.rs

```diff
@@ -394,6 +397,174 @@ pub enum LinkError {
     SyscallError(#[from] SyscallError),
 }
 
+/// A [`Link`] identifier, which may or may not be owned by aya.
+pub struct LinkId(u32);
+
+impl LinkId {
+    /// A wrapper for an arbitrary loaded link specified by its kernel id,
+    /// unsafe because there is no guarantee provided by aya that the link is
+    /// still loaded or even exists.
+    pub unsafe fn new(id: u32) -> Self {
```

Comment:
This isn't used interally by Aya.  It is pub so that a client that knows the kernel id for a link can create a LinkId that can be used by Aya.